src/game/SpellEffects.cpp

```diff
@@ -4285,7 +4285,7 @@ void Spell::EffectDismissPet(SpellEffectIndex /*eff_idx*/)
     if (!pet || !pet->isAlive())
         return;
 
-    pet->Unsummon(PET_SAVE_NOT_IN_SLOT, m_caster);
+    pet->Unsummon(PET_SAVE_AS_CURRENT, m_caster);
```

Comment:
You sure you don't wanna do a check for the type here as well and only do save as current on hunter pets and continue to do not in slot for the others. You tested to make sure it didn't break anything with dismissing warlock pets?